airbyte-integrations/bases/debezium-v1-9-2/src/main/java/io/airbyte/integrations/debezium/internals/PostgresConverter.java

```diff
@@ -57,14 +102,23 @@ private void registerText(final RelationalColumn field, final ConverterRegistrat
   }
 
   private void registerDate(final RelationalColumn field, final ConverterRegistration<SchemaBuilder> registration) {
+    final var fieldType = field.typeName();
+
     registration.register(SchemaBuilder.string().optional(), x -> {
       if (x == null) {
         return DebeziumConverterUtils.convertDefaultValue(field);
-      } else if (x instanceof PGInterval) {
-        return convertInterval((PGInterval) x);
-      } else {
-        return DebeziumConverterUtils.convertDate(x);
       }
+      return switch (fieldType.toUpperCase(Locale.ROOT)) {
+        case "TIMETZ" -> DateTimeConverter.convertToTimeWithTimezone(x);
+        case "TIMESTAMPTZ" -> DateTimeConverter.convertToTimestampWithTimezone(x);
+        case "TIMESTAMP" -> DateTimeConverter.convertToTimestamp(x);
+        // Debezium doesn't handle era indicators
+        // https://github.com/airbytehq/airbyte/issues/14590
+        case "DATE" -> DateTimeConverter.convertToDate(x);
+        case "TIME" -> DateTimeConverter.convertToTime(x);
+        case "INTERVAL" -> convertInterval((PGInterval) x);
```

Comment:
We have a debezium property which we can specify so that debezium can handle interval data type differently Please take a look at this and let me know what you think of it. https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-property-interval-handling-mode